pydantic/main.py

```diff
@@ -632,7 +632,12 @@ def copy(
         )
 
         # new `__fields_set__` can have unset optional fields with a set value in `update` kwarg
-        if update:
+        extra_ignore = self.__config__.extra == Extra.ignore
+        if update and extra_ignore:
```

Comment:
perhaps neater to have
```suggestion
        if update:
            if self.__config__.extra == Extra.ignore:
                ...
            else:
                fields_set = self.__fields_set__ | update.keys()      
```